lib/internal/streams/writable.js

```diff
@@ -463,12 +464,17 @@ function _write(stream, chunk, encoding, cb) {
     if (typeof chunk === 'string') {
       if ((state[kState] & kDecodeStrings) !== 0) {
         chunk = Buffer.from(chunk, encoding);
+        length = chunk.length;
         encoding = 'buffer';
+      } else {
+        length = Buffer.byteLength(chunk);
```

Comment:
Should we use the `encoding` argument instead of forcing UTF-8?

Reply:
Use the encoding argument